### PR TITLE
Optimize calendar reduce -every

### PR DESCRIPTION
**Pull request by davidvgalbraith**
if reducing -every :month: or -every :year:, send a date histogram
to ES with the appropriate interval

fixes https://github.com/juttle/juttle-elastic-adapter/issues/60

@demmer @VladVega

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,10 +23,5 @@ node_js:
     - '4.2'
     - '5.0'
 
-before_script:
-    - npm install -g jshint
-    - npm install -g jscs
-
 script:
-    - npm run -s lint
```
Comment (VladVega) [inline .travis.yml]: So @davidvgalbraith, did you get upset with jscs/lint yesterday? Or is there some logic behind removing this stuff in this PR?

**Comment by davidvgalbraith** [comment]
See the second commit message.

**Comment by VladVega** [comment]
Got it. So apparently you'll get the same thing with `eslint`: https://jslinterrors.com/dont-make-functions-within-a-loop. How about `_.each(new Array(count), function(v, k){})` for the outer loop?

**Comment by VladVega** [comment]
So why is only month and year? And not all intervals?

**Comment by VladVega** [comment]
Seems like a bunch of intervals will work: https://www.elastic.co/guide/en/elasticsearch/reference/2.0/search-aggregations-bucket-datehistogram-aggregation.html

**Comment by davidvgalbraith** [comment]
_.each(new Array is inaesthetic. I'll disable the rule when I have eslint. Single months and single years are the only calendar histogram a ES supports.

**Comment by VladVega** [comment]
One could maybe argue that it's also inaesthetic to have functions in a loop. Even though that particular one gets executed as a loop anyway. Anyway I'm going to let this go :)

**Comment by VladVega** [comment]
+1

**Pull request merged**